fontforgeexe/metricsview.c

```diff
@@ -5181,8 +5183,11 @@ MetricsView *MetricsViewCreate(FontView *fv,SplineChar *sc,BDFFont *bdf) {
     FontRequest rq;
     static GWindow icon = NULL;
     extern int _GScrollBar_Width;
-    // Max. glyphname length: 31, max. chars picked up: 15. 31*15 = 465
-    char buf[465], *pt;
+    // The maximum length of a glyph's name is 31 chars.
+#define MAXGLYPHNAME_LEN 31
+    unsigned int selectmax = fvmv_selectmax < 0 ? fv->b.sf->glyphcnt : fvmv_selectmax;
+    char buf[selectmax * MAXGLYPHNAME_LEN], *pt;
+    char titlebuf[50+strlen(fv->b.sf->fontname)+1];
```

Comment:
Let's avoid VLAs like this, just malloc it. This can easily lead to a stack overflow depending on how large this is.